lib/ui/window/pointer_data.h

```diff
@@ -9,6 +9,9 @@
 
 namespace flutter {
 
+// Must match the button constants in events.dart.
+const int64_t kPrimaryButton = 0x01;
```

Comment:
Why is this a free constant rather than an `enum class` like the others below? In my experience it's standard in C++ to use enums for flags.

It would also be good to just go ahead and add the other mouse button constants from events.dart at the same time so the list here matches, rather than being a subset.

Reply:
I'm not sure if it's ok to use enum as bitfield.

The reason I chose not to add other button constants is that only `kPrimaryButton` is used in code, since only it is used synthesis. When other buttons are used they will be directly copied.

I'll fix them if they're ok.

Reply:
> I'm not sure if it's ok to use enum as bitfield.

See https://github.com/flutter/engine/blob/master/lib/ui/semantics/semantics_node.h#L21 and later in the same file for example.

> only kPrimaryButton is used in code

We'll need all of them when it becomes part of the embedding API, since we'll need to translate each public constant to the corresponding internal constant.

Reply:
Note: Stuart and I discussed privately about whether to use enum types or int64_t. 

Pros of enum class:
- Type safe, such as preventing `kMousePrimaryButton | kStylusSecondaryButton`.
- Grouping comes more natively than explicit namespace.

Cons:
- Require to define `operator&` and `operator|`. It is magnified if we're to define all 4 sets of constants.
- Combined values no longer fall into enum options, which might be confusing.

The cost to implement all operators is considerable, while the benefit seems insignificant because they're not currently used in engine (also it is allowed in framework), I'm going forward with int64_t constants.